lib/middleware/website.js

```diff
@@ -82,6 +82,23 @@ exports = module.exports = () =>
 
       const key = err.detail.Key;
 
+      if (config.routingRules) {
+        for (const routingRule of config.routingRules) {
+          if (routingRule.shouldRedirect(key, 404)) {
```

Comment:
I attempted this before, because I agree it should be possible; however, I couldn't get a testcase working to simulate the behaviour. I can give it another shot, but if you have any tips on how to get the server into a state where it'll respond with a different status code that'd be very helpful.

Reply:
After a bit more experimenting with S3, I still can’t find an actual scenario where I can get it to redirect with anything other than a 404. S3 doesn’t do blanket redirections, I assume depending on where the error occurs will decide whether the routing rules kick in.

- S3 responds with a 403 when a file isn’t public accessible = No redirect
- S3 responds with a 405 when you try accessing a key with an unexpected method (e.g. a PUT) = No redirect

I’m a bit stumped. I can’t find any examples of people doing anything other than 404.

So I'm a bit wary of allowing any error to trigger a redirect in s3rver, as that's not the same behaviour as S3, but it's kinda hard to determine what actually is the expected behaviour! I'll add a comment for now.

Reply:
More experimenting yielded: `The provided HTTP error code (200) is not valid. Valid codes are 4XX or 5XX`, so I'll add this as an extra validation check too.